app/models/concerns/hashed_password_overrides.rb

```diff
@@ -0,0 +1,55 @@
+# In order to perform scrypt calculation of hashed password in a single
+# place for both password and PII encryption, we override
+# a few methods to build the encrypted_password via KeyMaker.
+#
+module HashedPasswordOverrides
+  extend ActiveSupport::Concern
+
+  attr_accessor :hashed_password
+
+  # hashed_password and unlock_password are new methods.
+  def unlock_password(password)
+    self.hashed_password = PasswordHasher.hash(password, authenticatable_salt)
+    key_maker.unlock(hashed_password, encryption_key)
+    hashed_password
+  end
+
+  def self.required_fields(_klass)
```

Comment:
I usually see all class methods defined above all instance methods in ruby classes, the exception being `def initialize` at the top if present. 

Is there a reason to not put this class method before the rest of these methods?